viewing_party/party.py

```diff
@@ -1,23 +1,168 @@
+        movies.update({"genre": genre, "rating": rating, "title": title})
+        return movies
+    
+    else:
+        return None
 
+def add_to_watched(user_data, movie):
+    user_data.update({"watched": [movie]})
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data.update({"watchlist": [movie]})
+    return user_data
+
+def watch_movie(user_data, title):
+
+    #work through the whole watchlist; copy it first so no DANGERZONE/index/key errors
+    copy_list = user_data["watchlist"].copy()
+    for i in range(len(copy_list)):
+
+    #find the title of the movie in the watchlist
+        if copy_list[i]["title"] == title:
+
+            #if the title is in the watchlist, move it to watched then delete
+            user_data["watched"].append(copy_list[i])
+            del user_data["watchlist"][i]
+
+    return user_data
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    total = 0
+
+    #make sure the dictionary isn't empty
+    if user_data["watched"]:
+
+        #iterate over the watched list and add averages to total
+        for i in range(len(user_data["watched"])):
+            total += user_data["watched"][i]["rating"]
+
+    #if nothing in the watched list, return 0
+    else:
+        return 0
+
+    #calculate the average and return it
+    average = total/len(user_data["watched"])
+    return average
 
+def genres(user_data):
+    #set up set, dictionary for the count
+    genres = set()
+    
+    #make a set of genres, and return it as a list
+    for i in range(len(user_data["watched"])):
+        genres.add(user_data["watched"][i]["genre"])
+    return list(genres)
+
+def genre_count(user_data):
+    #set up set, dictionary for the count
+    user_genres = genres(user_data)
+    genre_count = {}
+
+    #iterate over the genres and add a count to a dictionary
+    for i in range(len(user_data["watched"])):
+        count = 0
+        for genre in user_genres:
+            if user_data["watched"][i]["genre"] == genre:
+                count += 1
+                genre_count[genre] = count
+        return genre_count
+
+def get_most_watched_genre(user_data):
+    #make sure the watched list has data
+    if user_data["watched"]:
+        watched_genre_count = genre_count(user_data)
+        most_watched = max(watched_genre_count, key=watched_genre_count.get)
+        return most_watched
+    
+    #if no data, return None
+    else:
+        return None
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_friend_data(user_data):
+    #make comprehensive list of all friend data
+    friend_data = []
+    for i in range(len(user_data["friends"])):
+        for j in range(len(user_data["friends"][i]["watched"])):
+            friend_data.append(user_data["friends"][i]["watched"][j])

```

Comment:
Parsing the nested functions and data here is a little hard to read. Similar to the feedback above, I would suggest converting this to loop directly over the lists, but if we did want to loop over ranges, I would declare some variables to help reduce repeated access to nested things and make it easier to read: 
```suggestion
    for i in range(len(user_data["friends"])):
        friend_watched_list = user_data["friends"][i]["watched"]
        for j in range(len(friend_watched_list)):
            friend_data.append(friend_watched_list[j])
```